### PR TITLE
ENR records limited to 300 bytes

### LINKED ISSUE #3708: ENR records should be limited to 300 bytes.
**Describe the bug**
Maximum size of an encoded node record is 300 bytes.

### PR DESCRIPTION
**Pull request by smartprogrammer93**
Closes #3708

## Changes:
-throws when enr record rlp recieved is bigger than 300 bytes

## Types of changes

What types of changes does your code introduce?
_Put an `x` in the boxes that apply_

- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation Update
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [x] Other (please describe): Just an extra validation check on ENR records received

## Testing
**Requires testing**

- [x] Yes
- [ ] No

**In case you checked yes, did you write tests??**

- [x] Yes
- [ ] No

### CODE REVIEW

File: src/Nethermind/Nethermind.Network.Enr/NodeRecordSigner.cs [hunk 1]
```diff
@@ -55,7 +55,8 @@ public NodeRecord Deserialize(RlpStream rlpStream)
     {
         int startPosition = rlpStream.Position;
         int recordRlpLength = rlpStream.ReadSequenceLength();
-
+        if (recordRlpLength > 300)
```
Comment (LukaszRozmej) [inline src/Nethermind/Nethermind.Network.Enr/NodeRecordSigner.cs:58]: Just thinking shouldn't this be `RlpException` at this level and potentially packed to `NetworkingException` on different level?

Comment (smartprogrammer93) [reply src/Nethermind/Nethermind.Network.Enr/NodeRecordSigner.cs:58]: i mean, the RLP as not invalid. it is just too big to be a valid ENR, right?

Comment (LukaszRozmej) [reply src/Nethermind/Nethermind.Network.Enr/NodeRecordSigner.cs:58]: That is a good philosophical question

**Review by LukaszRozmej — APPROVED** [review]

File: src/Nethermind/Nethermind.Network.Enr/NodeRecordSigner.cs (hunk 1 above)
Comment (LukaszRozmej) [reply src/Nethermind/Nethermind.Network.Enr/NodeRecordSigner.cs:58]: Where is this called?

**Pull request merged**